pkg/status/conditions.go

```diff
@@ -0,0 +1,200 @@
+var clock kubeclock.Clock = &kubeclock.RealClock{}
+
+// ConditionType is the type of the condition and is typically a CamelCased
+// word or short phrase.
+//
+// Condition types should indicate state in the "abnormal-true" polarity. For
+// example, if the condition indicates when a policy is invalid, the "is valid"
+// case is probably the norm, so the condition should be called "Invalid".
+type ConditionType string
+
+// ConditionReason is intended to be a one-word, CamelCase representation of
+// the category of cause of the current status. It is intended to be used in
+// concise output, such as one-line kubectl get output, and in summarizing
+// occurrences of causes.
+type ConditionReason string
+
+// Condition represents an observation of an object's state. Conditions are an
+// extension mechanism intended to be used when the details of an observation
+// are not a priori known or would not apply to all instances of a given Kind.
+//
+// Conditions should be added to explicitly convey properties that users and
+// components care about rather than requiring those properties to be inferred
+// from other observations. Once defined, the meaning of a Condition can not be
+// changed arbitrarily - it becomes part of the API, and has the same
+// backwards- and forwards-compatibility concerns of any other part of the API.
+type Condition struct {
+	Type               ConditionType          `json:"type"`
+	Status             corev1.ConditionStatus `json:"status"`
+	Reason             ConditionReason        `json:"reason,omitempty"`
+	Message            string                 `json:"message,omitempty"`
+	LastTransitionTime metav1.Time            `json:"lastTransitionTime,omitempty"`
+}
+
+// IsTrue Condition whether the condition status is "True".
+func (c Condition) IsTrue() bool {
+	return c.Status == corev1.ConditionTrue
+}
+
+// IsFalse returns whether the condition status is "False".
+func (c Condition) IsFalse() bool {
+	return c.Status == corev1.ConditionFalse
+}
+
+// IsUnknown returns whether the condition status is "Unknown".
+func (c Condition) IsUnknown() bool {
+	return c.Status == corev1.ConditionUnknown
+}
+
+// DeepCopy returns a deep copy of the condition
+func (c *Condition) DeepCopy() *Condition {
+	if c == nil {
+		return nil
+	}
+	out := *c
+	return &out
+}
+
+// Conditions is a set of Condition instances.
+//
+// +kubebuilder:validation:Type=array
+type Conditions map[ConditionType]Condition
+
+// NewConditions initializes a set of conditions with the given list of
+// conditions.
+func NewConditions(conds ...Condition) Conditions {
+	conditions := Conditions{}
+	for _, c := range conds {
+		conditions.SetCondition(c)
+	}
+	return conditions
+}
+
+// IsTrue searches the set of conditions for a condition with the given
+// ConditionType. If found, it returns `condition.IsTrue()`. If not found,
+// it returns false.
+func (conditions Conditions) IsTrue(t ConditionType) bool {
```

Comment:
`IsTrueFor(t)` would make more sense. Ditto for other Boolean methods accepting a type argument.